lib/nostr.js

```diff
@@ -16,24 +20,52 @@ export const DEFAULT_CROSSPOSTING_RELAYS = [
   'wss://nostr.mutinywallet.com/',
   'wss://relay.mutinywallet.com/'
 ]
+export const DEFAULT_NIP46_RELAYS = [
+  'wss://relay.damus.io/',
+  'wss://relay.snort.social/'
+]
 
 export class Relay {
-  constructor (relayUrl) {
-    const ws = new WebSocket(relayUrl)
+  constructor (relayUrl, reconnectOnClose = false) {
+    this.url = relayUrl
+    this.subs = []
+    this.reconnectOnClose = reconnectOnClose
+    this._closed = false
+    this.error = undefined
+    this.reconnect()
+  }
 
-    ws.onmessage = function (msg) {
+  reconnect () {
+    this._closed = false
+    const ws = new WebSocket(this.url)
+    this.ws = ws
+
+    ws.onmessage = (msg) => {
       const [type, notice] = JSON.parse(msg.data)
       if (type === 'NOTICE') {
         console.log('relay notice:', notice)
       }
     }
 
-    ws.onerror = function (err) {
+    ws.onerror = (err) => {
       console.error('websocket error: ' + err)
       this.error = err
     }
 
-    this.ws = ws
+    ws.onclose = () => {
+      if (this.reconnectOnClose && !this._closed) {
+        setTimeout(() => {
+          this.reconnect()
+        }, 5000)
+      }
+    }
+
+    ws.onopen = () => {
+      // Resubscribe to all subscriptions on reconnect
+      for (const sub of this.subs) {
+        ws.send(JSON.stringify(['REQ', sub.id, ...sub.filters]))
+      }
```

Comment:
Should this run even if reconnecting was not enabled?

Reply:
i could add an if there if you want, but unless there is a reconnection this.subs will be empty so the for loop will exit immediately